listenbrainz/webserver/static/js/src/playlists/Playlist.tsx

```diff
@@ -560,58 +560,26 @@ export default class PlaylistPage extends React.Component<
       );
       return;
     }
-    const { title, annotation, identifier } = playlist;
-    const customFields = getPlaylistExtension(playlist);
     this.setState({ loading: true });
     try {
-      const newPlaylist: SpotifyPlaylistObject =
-        this.spotifyPlaylist ||
-        (await this.SpotifyAPIService.createPlaylist(
-          title,
-          customFields?.public,
-          `${annotation}
-          Exported from ListenBrainz playlist ${identifier}`
-        ));
-
-      if (!this.spotifyPlaylist) {
-        // Store the playlist ID, in case something goes wrong we don't recreate another playlist
-        this.spotifyPlaylist = newPlaylist;
-      }
-
-      const spotifyURIs = await this.SpotifyAPIService.searchForSpotifyURIs(
-        playlist.track
-      );
-      await this.SpotifyAPIService.addSpotifyTracksToPlaylist(
-        newPlaylist.id,
-        spotifyURIs
+      const playlistId = getPlaylistId(playlist);
+      const result = await this.APIService.exportPlaylistToSpotify(
+        currentUser.auth_token,
+        playlistId
       );
-      const playlistLink = `https://open.spotify.com/playlist/${newPlaylist.id}`;
+      const { external_url } = result;
       newAlert(
         "success",
         "Playlist exported to Spotify",
         <>
           Successfully exported playlist:{" "}
-          <a href={playlistLink} target="_blank" rel="noopener noreferrer">
-            {playlistLink}
+          <a href={external_url} target="_blank" rel="noopener noreferrer">
+            {external_url}
           </a>
-          {spotifyURIs.length !== playlist.track.length && (
-            <b>
-              <br />
-              {playlist.track.length - spotifyURIs.length} tracks were not found
-              on Spotify, and consequently skipped.
-            </b>
-          )}
         </>
       );
     } catch (error) {
```

Comment:
The invalid scopes error is handled internally by the endpoint as it can refresh tokens so that error should never reach the frontend. Even if it bubbles up, it'll appear as a 500 error.

Reply:
Fixed,